### PR TITLE
add igvm_locked check

### PR DESCRIPTION
**Pull request by Jowe1999**
add igvm_locked check

### CODE REVIEW

File: src/check_igvm_locked.py [hunk 1]
```diff
@@ -0,0 +1,95 @@
+from adminapi.dataset import Query
```
Comment (Woellchen) [inline src/check_igvm_locked.py]: copyright and documentation missing

**Comment by Woellchen** [comment]
`flake8` currently reports 24 errors. please fix them

File: src/check_igvm_locked.py [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+from datetime import datetime, timezone, timedelta
+from subprocess import Popen, PIPE, DEVNULL
+
+import argparse
+
+
+
+
+def main():
+
```
Comment (Woellchen) [inline src/check_igvm_locked.py]: please remove useless newline

File: src/check_igvm_locked.py [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+    parser = argparse.ArgumentParser(description='nagios check for long running igvm_locked attribute')
+    parser.add_argument("monitoring_master", type=str)
+    parser.add_argument("--time_in_minutes", type=int, default=480)
```
Comment (Woellchen) [inline src/check_igvm_locked.py]: a nice cli flag would rather look like `--time-in-minutes`

File: src/check_igvm_locked.py [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+    nsca_output = ""
+    for host in hosts_locked:
+        nsca_output += ('{}\tigvm_locked\t{}\tWARNING - IGVM-locked longer than {}h {}m\x17'
+            .format(host['hostname'],1,int(max_minutes/60),int(max_minutes - 60 * (max_minutes/60))))
+
+    for host in hosts_not_locked:
+        nsca_output += ('{}\tigvm_locked\t{}\tOK\x17'
+            .format(host['hostname'],0))
+
+    return(nsca_output)
```
Comment (Woellchen) [inline src/check_igvm_locked.py]: remove useless parenthesis please

File: src/check_igvm_locked.py [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+    result = True
+    nsca = Popen(
+        [
+            '/usr/sbin/send_nsca',
+            '-H', host,
+            '-c', '/etc/send_nsca.cfg',
+        ],
+        stdin=PIPE,
+        stdout=DEVNULL,
+        stderr=DEVNULL
```
Comment (Woellchen) [inline src/check_igvm_locked.py]: lets have trailing commas. that way it is consistent and doesn't bloat the diff next time we change something here

File: src/check_igvm_locked.py [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+    )
+    nsca.communicate(nsca_output.encode())
+    returncode = nsca.wait()
+    if returncode > 0:
+        result = False
+
+    return result
```
Comment (Woellchen) [inline src/check_igvm_locked.py]: how about `return not bool(returncode)`

File: src/check_igvm_locked.py [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+
+
+def console_out(hosts_locked, max_minutes):
+    count_locked_servers = 0
+    locked_hosts = ""
+    for host in hosts_locked:
+        #locked_hosts = str(locked_hosts) + str(host['hostname'] + ":" + str(host['igvm_locked']) + "\n")
```
Comment (Woellchen) [inline src/check_igvm_locked.py]: please remove dead/commented code if it is not needed

File: src/check_igvm_locked.py [hunk 8]
```diff
@@ -0,0 +1,95 @@
...
+        locked_hosts += "{}:{}\n".format(host['hostname'], host['igvm_locked'])
+        count_locked_servers += 1
+
+    if locked_hosts == "":
+       print("No igvm_locked Servers !!!")
+    else:
+       #print("This servers are locked longer than " + str(int(max_minutes/60)) + "h and " + str(max_minutes - (int(max_minutes/60) * 60)) + "m :" + str(locked_hosts))
```
Comment (Woellchen) [inline src/check_igvm_locked.py]: please remove dead/commented code if it is not needed

File: src/check_igvm_locked.py [hunk 9]
```diff
@@ -0,0 +1,95 @@
...
+    hosts_locked = []
+    for host in hosts:
+        if not host['igvm_locked']:
+            hosts_not_locked.append(host)
+        elif (datetime.now(timezone.utc) - host['igvm_locked']) > timedelta(minutes=max_minutes):
+            hosts_locked.append(host)
+        else:
+            hosts_not_locked.append(host)
+
+    console_out(hosts_locked, max_minutes)
```
Comment (Woellchen) [inline src/check_igvm_locked.py]: is this a pure nsca check? why do we need to print something to the console then? and if we need to it should be the exact same text that is sent to nsca (`nagios_create`)

File: src/check_igvm_locked.py [hunk 10]
```diff
@@ -0,0 +1,95 @@
...
+    nsca_out = nagios_create(hosts_locked, hosts_not_locked, max_minutes)
+    nagios_send(master, nsca_out)
+
+def nagios_create(hosts_locked, hosts_not_locked, max_minutes):
+    nsca_output = ""
+    for host in hosts_locked:
+        nsca_output += ('{}\tigvm_locked\t{}\tWARNING - IGVM-locked longer than {}h {}m\x17'
```
Comment (Woellchen) [inline src/check_igvm_locked.py]: i am not sure if this is an acceptable nsca output string. maybe @lamaral can enlighten us

Comment (lamaral) [reply src/check_igvm_locked.py]: Weird as it looks, NSCA uses it like that. 
```
Service Checks:
<host_name>[tab]<svc_description>[tab]<return_code>[tab]<plugin_output>[newline]

Host Checks:
<host_name>[tab]<return_code>[tab]<plugin_output>[newline]

When submitting multiple simultaneous results, separate each set with the ETB
character (^W or 0x17)
```

File: src/check_igvm_locked.py [hunk 11]
```diff
@@ -0,0 +1,95 @@
...
+
+    parser = argparse.ArgumentParser(description='nagios check for long running igvm_locked attribute')
+    parser.add_argument("monitoring_master", type=str)
+    parser.add_argument("--time_in_minutes", type=int, default=480)
+    args = parser.parse_args()
+
+    max_minutes = args.time_in_minutes
+    master = args.monitoring_master
+
+    hosts = Query({'servertype': Any('hypervisor', 'vm'), 'no_monitoring': False, 'state': Not('retired'), 'hostname': 'luiz.test.ig.local'}, ['igvm_locked', 'hostname'])
```
Comment (lamaral) [inline src/check_igvm_locked.py]: Please adjust this query, it still contains test data.

File: src/check_igvm_locked.py [hunk 12]
```diff
@@ -0,0 +1,95 @@
...
+    if locked_hosts == "":
+       print("No igvm_locked Servers !!!")
+    else:
+       #print("This servers are locked longer than " + str(int(max_minutes/60)) + "h and " + str(max_minutes - (int(max_minutes/60) * 60)) + "m :" + str(locked_hosts))
+        hours = int(max_minutes/60)
+        minutes = max_minutes - (int(max_minutes/60) * 60)
+
+
+
+        print("({}) server/s are/is locked longer than {}h and {}m : \n{}".format(count_locked_servers, hours, minutes, locked_hosts))
```
Comment (lamaral) [inline src/check_igvm_locked.py]: This line is too long. It can be broken down into multiple lines.

File: src/check_igvm_locked.py (hunk 9 above)
Comment (lamaral) [reply src/check_igvm_locked.py]: We could maybe toggle the output with a `-v` flag?

File: src/check_igvm_locked.py (hunk 10 above)
Comment (Woellchen) [reply src/check_igvm_locked.py]: but now it looks like this:
> (1) server/s are/is locked longer than 1h and 0m :
>***:####-##-## ##:##:##+##:##
>
>***	igvm_locked	1	WARNING - IGVM-locked longer than bla
is this still correct?!

Comment (Woellchen) [reply src/check_igvm_locked.py]: ok scratch that. i had tested it locally and it was just because there were other prints to the console

**Comment by Jowe1999** [comment]
changed the requested comments

File: src/check_igvm_locked.py [hunk 13]
```diff
@@ -0,0 +1,117 @@
...
+import argparse
+
+
+def main():
+    parser = argparse.ArgumentParser(
+        description='nagios check for long running igvm_locked attribute',
+    )
+    parser.add_argument("monitoring_master", type=str)
+    parser.add_argument("--time-in-minutes", type=int, default=480)
+    parser.add_argument("-v")
```
Comment (lamaral) [inline src/check_igvm_locked.py]: This should be a boolean argument. The way it is, you need to give data to the argument.

File: src/check_igvm_locked.py [hunk 14]
```diff
@@ -0,0 +1,117 @@
...
+import argparse
+
+
+def main():
+    parser = argparse.ArgumentParser(
+        description='nagios check for long running igvm_locked attribute',
+    )
+    parser.add_argument("monitoring_master", type=str)
+    parser.add_argument("--time-in-minutes", type=int, default=480)
+    parser.add_argument("-v", type=bool, default=False)
```
Comment (lamaral) [inline src/check_igvm_locked.py]: This still requires you to do something like `python3 check.py --time-in-minutes 60 -v True localhost`.

Try it like this:
```
parser.add_argument('-v', action='store_true')
```

Comment (Jowe1999) [reply src/check_igvm_locked.py]: okey thx

**Comment by Woellchen** [comment]
it is fine now

**Pull request merged**